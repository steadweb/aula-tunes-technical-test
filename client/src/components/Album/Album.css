.container.blur .album-container .album {
  filter: blur(4px);
  -webkit-filter: blur(4px);
}

.album-container .album {
  cursor: pointer;
}

.album-container .album:hover {
  opacity: 0.4;
  transition: opacity 0.3s;
}
