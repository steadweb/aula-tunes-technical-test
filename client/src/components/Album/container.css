.album-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  flex-grow: 1;
}
