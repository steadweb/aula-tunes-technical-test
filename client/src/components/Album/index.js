import React from 'react';
import './Album.css';

function Album({ artworkUrl100, onClick }) {
  return (
    <>
      <div className='album' onClick={onClick}>
        <img src={artworkUrl100} alt='Album Cover' />
      </div>
    </>
  );
}

export default Album;
