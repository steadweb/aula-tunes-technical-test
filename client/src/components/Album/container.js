import React from 'react';
import Album from './index';
import './container.css';

function AlbumContainer({ feed, setActiveAlbum }) {
  return (
    <div className='album-container'>
      {feed.results.map((feedItem) => (
        <Album
          key={feedItem.id}
          {...feedItem}
          onClick={() => setActiveAlbum(feedItem)}
        />
      ))}
    </div>
  );
}

export default AlbumContainer;
