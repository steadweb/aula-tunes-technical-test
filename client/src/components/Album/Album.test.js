import React from 'react';
import renderer from 'react-test-renderer';
import Album from './index';

describe('Album', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Album artworkUrl100='/path/to/image' onClick={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
