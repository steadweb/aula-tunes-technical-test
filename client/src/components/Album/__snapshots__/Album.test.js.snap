// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album renders correctly 1`] = `
<div
  className="album"
  onClick={[Function]}
>
  <img
    alt="Album Cover"
    src="/path/to/image"
  />
</div>
`;
